### PR TITLE
Add doc for Navigation Debug PerformanceMonitor

### PR DESCRIPTION
**Pull request by smix8**
Adds documentation for Navigation Debug PerformanceMonitor.

### CODE REVIEW

File: tutorials/navigation/navigation_debug_tools.rst [hunk 1]
```diff
@@ -59,3 +59,31 @@ The connections can be made visible through geometry with the navigation debug `
...
     Edge connections are only visible when the NavigationServer is active.
+
+Debug Performance
+-----------------
+
+To measure NavigationServer performance a dedicated monitor exists that can be found within the Editor Debugger under `Debugger->Monitors->NavigationProcess`.
+
+.. image:: img/navigation_debug_performance1.png
+
+NavigationProcess tells how long the NavigationServer spend updating its internals this update frame in milliseconds.
```
Comment (mhilbrunner) [inline tutorials/navigation/navigation_debug_tools.rst:70]:
```suggestion
NavigationProcess shows how long the NavigationServer spend updating its internals this update frame in milliseconds.
```

Not sure about this one

**Review by mhilbrunner — COMMENTED** [review]
Looks good to me so far, didn't spot anything. Please convert the images to WebP!

**Comment by smix8** [comment]
Changed images to WebP.

File: tutorials/navigation/navigation_debug_tools.rst [hunk 2]
```diff
@@ -59,3 +59,31 @@ The connections can be made visible through geometry with the navigation debug `
...
+Note that since the NavigationServer process update happens in the middle of the physics update an increase in NavigationProcess will automatically increase PhysicsProcess by the same amount.
+
+Navigation also provides more detailed statistics about the current navigation related objects and navigation map composition on the NavigationServer.
+
+.. image:: img/navigation_debug_performance2.webp
+
+Navigation statistics shown here can not be judged as good or bad for performance as it depends entirely on the project what can be considered as reasonable or horribly excessive.
+
+Navigation statistics help with identifying performance bottlenecks that are less obvious because the source might not always have a visible representation.
+E.g. pathfinding performance issues created by overly detailed navigation meshes with thousand of edges / polygons or problems caused by procedual navigation gone wrong.
```
Comment (skyace65) [inline tutorials/navigation/navigation_debug_tools.rst:89]:
```suggestion
E.g. pathfinding performance issues created by overly detailed navigation meshes with thousand of edges / polygons or problems caused by procedural navigation gone wrong.
```

File: tutorials/navigation/navigation_debug_tools.rst [hunk 3]
```diff
@@ -59,3 +59,31 @@ The connections can be made visible through geometry with the navigation debug `
...
     Edge connections are only visible when the NavigationServer is active.
+
+Debug Performance
+-----------------
+
+To measure NavigationServer performance a dedicated monitor exists that can be found within the Editor Debugger under `Debugger->Monitors->NavigationProcess`.
+
+.. image:: img/navigation_debug_performance1.webp
+
+NavigationProcess shows how long the NavigationServer spend updating its internals this update frame in milliseconds.
```
Comment (skyace65) [inline tutorials/navigation/navigation_debug_tools.rst:70]:
```suggestion
NavigationProcess shows how long the NavigationServer spends updating its internals this update frame in milliseconds.
```

File: tutorials/navigation/navigation_debug_tools.rst [hunk 4]
```diff
@@ -59,3 +59,31 @@ The connections can be made visible through geometry with the navigation debug `
...
+NavigationProcess works similar to Process for visual frame rendering and PhysicsProcess for collision and fixed updates.
+
+NavigationProcess accounts for all updates to ``navigation maps``, ``navigation regions`` and ``navigation agents`` as well as the entire ``avoidance calculations`` for the update frame.
```
Comment (skyace65) [inline tutorials/navigation/navigation_debug_tools.rst:73]: Is ``avoidance calculations`` in this sentence referring to a specific thing, like a property named ``avoidance calculations``? or does this just mean every avoidance calculation? If this means every avoidance calculation and not a singular thing "the entire" should be replaced with "all the" in this sentence,

**Review by skyace65 — CHANGES_REQUESTED** [review]
Looks good, just needs a few changes.

File: tutorials/navigation/navigation_debug_tools.rst (hunk 4 above)
Comment (smix8) [reply tutorials/navigation/navigation_debug_tools.rst]: means every avoidance calculation

**Review by skyace65 — APPROVED** [review]
looks good now!

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thank you! 🎉

**Pull request merged**